reframe/core/variables.py

```diff
@@ -47,7 +48,7 @@ def __init__(self, *args, **kwargs):
         self._default_value = kwargs.pop('value', Undefined)
 
         if not issubclass(field_type, fields.Field):
-            raise ValueError(
+            raise ReframeSyntaxError(
```

Comment:
Passing an invalid descriptor here as `fields` is not a syntax error. If I get a syntax error from this, as a user, I'd interpret that I have some syntax errors in this API call, but the API call is sound. Instead, the problem is just that the values that I'm passing have the wrong type. So actually, perhaps a `TypeError` would make even more sense here.

Reply:
I agree with that and I had doubts about this and a couple of other errors. The problem is, however, that if we raise any builtin exception, then we get a full stack trace and loading is stopped, because we assume that this is a framework error and not a user one, as it actually is.

Reply:
To me, this seems something that should be addressed at the loader level, and not here. I also think hat we need some rationale to define what is considered a user error and what is not. Our `TypedField` descriptors also throw builtin errors when the user sets them to a wrong type. The only difference is that such errors get thrown later on.
In any case, the `field` argument here is not documented, so I still think this should be a `TypeError`.

Reply:
Ok, I thought it was documented. I can then change it back to `TypeError`. As of the user error, my proposal for the rational is in the description of this PR. And indeed it's the loader that handles them by issuing a warning. The problem with these errors is that you can't easily track their origin back to the user, especially with all the metaclass magic. So for the metaclass magic we need to issue a specific error that can help the loader differentiate.

Reply:
Actually, what about a try/except block in the `load_from_file` method from the `RegressionCheckLoader`, and we catch all Exceptions there? We could then just raise a warning saying _"Could not load checks from file XXXX due to errors"_